lib/crypto/PairwiseKey.ts

```diff
@@ -221,60 +193,42 @@ export default class PairwiseKey {
     let keySize = minimumKeySize;
     if (algorithm.modulusLength) {
       keySize = algorithm.modulusLength;
-      if (keySize && keySize < minimumKeySize) {
-        keySize = minimumKeySize;
-      } else {
-        keySize = minimumKeySize;
-      }
     }
 
     // Get peer id
     let peerId = Buffer.from(this._peerId);
-    let pBase: Buffer;
-    let qBase: Buffer;
+
     // Get pbase
-    return this.generateDeterministicNumberForPrime(crypto, keySize / 2, didMasterKey, peerId).then((pbase: Buffer) => {
-      pBase = pbase;
-      return pbase;
-    }).then((pbase: Buffer) => {
+    let pBase: Buffer = await this.generateDeterministicNumberForPrime(crypto, keySize / 2, didMasterKey, peerId);
     // Get qbase
-      return this.generateDeterministicNumberForPrime(crypto, keySize / 2, pbase, peerId).then((qbase: Buffer) => {
-        qBase = qbase;
-        return qbase;
-      }).then(() => {
-        let p = this.getPrime(pBase);
-        let q = this.getPrime(qBase);
+    let qBase: Buffer = await this.generateDeterministicNumberForPrime(crypto, keySize / 2, pBase, peerId);
+    let p = this.getPrime(pBase);
+    let q = this.getPrime(qBase);
 
           // compute key components
-        let modulus = p.multiply(q);
-        let pMinus = p.subtract(bigInt.one);
-        let qMinus = q.subtract(bigInt.one);
-        let phi = pMinus.multiply(qMinus);
-        let e = bigInt(65537);
-        let d = e.modInv(phi);
-        let dp = d.mod(pMinus);
-        let dq = d.mod(qMinus);
-        let qi = q.modInv(p);
-        let jwk = {
-          kty: 'RSA',
-          use: keyUse.toString(),
-          e: this.toBase(e),
-          n: this.toBase(modulus),
-          d: this.toBase(d),
-          p: this.toBase(p),
-          q: this.toBase(q),
-          dp: this.toBase(dp),
-          dq: this.toBase(dq),
-          qi: this.toBase(qi)
-        };
+    let modulus = p.multiply(q);
```

Comment:
FNT (for next time): Use const rather than let for variables that will never change once initialized.